pkg/parse/run.go

```diff
@@ -88,11 +98,30 @@ func Run(ctx context.Context, p Parser, nsControllerState *namespacecontroller.S
 	nsEventTimer := time.NewTimer(nsEventPeriod)
 	defer nsEventTimer.Stop()
 
-	state := &reconcilerState{
-		backoff:     defaultBackoff(),
-		retryTimer:  retryTimer,
-		retryPeriod: opts.RetryPeriod,
+	var state *reconcilerState
+	var runFn RunFn
+
+	if runOpts != nil {
```

Comment:
to simplify this method, I'd recommend using a DefaultRunOpts to build the RunOps. Then make the RunOpts a non-pointer in the func receiver so it's required and have the caller call DefaultRunOpts to build the ops when executing the func.

Reply:
Done